### PR TITLE
Prow: update secrets population

### PR DESCRIPTION
**Pull request by Halamix2**
**Description**

Changes proposed in this pull request:

- remove references to `sap-kyma-secrets` bucket
- replace `secretspopulator` with External Secrets Syncer
- update documentation

**Related issue(s)**

### CODE REVIEW

File: docs/prow/obligatory-security-measures.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Read about the obligatory security measures to take on a regular basis and when
 
 ## Change Secrets regularly
 
-All Secret used in the Prow production cluster must be changed every six months. Follow [Prow secret management](./prow-secrets-management.md) to create a new key ring and new Secrets. Then, use [Secrets populator](./../../development/tools/cmd/secretspopulator/README.md) to change all Secrets in the Prow cluster.
+All Secret used in the Prow production cluster must be changed every six months. Follow [Prow secret management](./prow-secrets-management.md) to create a new key ring and new Secrets. Then, [External Secrets Syncer](../../prow/workload-cluster/README.md) will change all Secrets in the Prow cluster automatically.
```
Comment (mmitoraj) [inline docs/prow/obligatory-security-measures.md:7]:
```suggestion
All Secrets used in the Prow production cluster must be changed every six months. Follow the [Prow secret management](./prow-secrets-management.md) guidelines to create a new key ring and new Secrets. Once the key ring and new Secrets are ready, the [External Secrets Syncer](../../prow/workload-cluster/README.md) changes all Secrets in the Prow cluster automatically.
```
The link ->  [External Secrets Syncer](../../prow/workload-cluster/README.md) doesn't point the External Secrets Syncer

File: docs/prow/obligatory-security-measures.md [hunk 2]
```diff
@@ -22,4 +22,4 @@ Remove the person from the `kyma-prow` Google project immediately. Follow [this]
 
 ### Change Secrets
 
-Change all Secrets that were valid when the person was a project member. Follow [Prow secret management](./prow-secrets-management.md) to create a new key ring and new Secrets. Then, use [Secrets populator](./../../development/tools/cmd/secretspopulator/README.md) to change all Secrets in the Prow cluster.
+Change all Secrets that were valid when the person was a project member. Follow [Prow secret management](./prow-secrets-management.md) to create a new key ring and new Secrets. Then, [External Secrets Syncer](../../prow/workload-cluster/README.md) will change all Secrets in the Prow cluster automatically.
```
Comment (mmitoraj) [inline docs/prow/obligatory-security-measures.md:25]:
```suggestion
Change all Secrets that were valid when the person was a project member. Follow the [Prow secret management](./prow-secrets-management.md) guidelines to create a new key ring and new Secrets. Once the key ring and new Secrets are ready, the [External Secrets Syncer](../../prow/workload-cluster/README.md) changes all Secrets in the Prow cluster automatically.
```
The link ->  [External Secrets Syncer](../../prow/workload-cluster/README.md) doesn't point the External Secrets Syncer

File: docs/prow/prow-installation-on-forks.md [hunk 3]
```diff
@@ -106,13 +106,9 @@ Follow these steps to install Prow:
...
 1. Export these environment variables:
 
-- **BUCKET_NAME** is a GCS bucket in the Google Cloud project that stores Prow Secrets.
-- **KEYRING_NAME** is the KMS key ring.
-- **ENCRYPTION_KEY_NAME** is the key name in the key ring that is used for data encryption.
 - **KUBECONFIG** is a path to a `kubeconfig` file.
 - **PROJECT** is a GCP project name.
-- **GOOGLE_APPLICATION_CREDENTIALS** is a path to a service account file. This service account requires KMS and storage roles.  
-The account files are encrypted with the **ENCRYPTION_KEY_NAME** key from **KEYRING_NAME** and are stored in **BUCKET_NAME**.
+- **GOOGLE_APPLICATION_CREDENTIALS** is a path to a service account file. This service account requires read access to secretes stored in Secret Manager.
```
Comment (mmitoraj) [inline docs/prow/prow-installation-on-forks.md:111]:
```suggestion
- **GOOGLE_APPLICATION_CREDENTIALS** is a path to a service account file. This service account requires read access to Secrets stored in the Secret Manager.
```

**Review by mmitoraj — CHANGES_REQUESTED** [review]

**Review by mmitoraj — APPROVED** [review]

**Review by adamwalach — APPROVED** [review]

**Pull request merged**